File: python/45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 0
        
        jumps = 1
        curr_end = nums[0]
        next_end = nums[0]
        
        for i in range(1, len(nums)):
            if i > curr_end:
                jumps += 1
                curr_end = next_end
            
            next_end = max(next_end, i + nums[i])
        
        return jumps

File: C#/45.cs
public class Solution {
    public int Jump(int[] nums) {
        int n = nums.Length;
        int jumps = 0;
        int currEnd = 0;
        int currFarthest = 0;
        
        for (int i = 0; i < n - 1; i++) {
            currFarthest = Math.Max(currFarthest, i + nums[i]);
            if (i == currEnd) {
                jumps++;
                currEnd = currFarthest;
            }
        }
        
        return jumps;
    }
}